crates/wasi-c2/src/virt/pipe.rs

```diff
@@ -0,0 +1,373 @@
+// This is mostly stubs
+#![allow(unused_variables, dead_code)]
+//! Virtual pipes.
+//!
+//! These types provide easy implementations of `WasiFile` that mimic much of the behavior of Unix
+//! pipes. These are particularly helpful for redirecting WASI stdio handles to destinations other
+//! than OS files.
+//!
+//! Some convenience constructors are included for common backing types like `Vec<u8>` and `String`,
+//! but the virtual pipes can be instantiated with any `Read` or `Write` type.
```

Comment:
Probably a dumb question, but I wonder if there is a reason to have separate structs for read and write (`ReadPipe` and `WritePipe`) rather than a single bi-directional pipe, something like below?
https://github.com/ueno/wasmtime/blob/wip/dueno/wasi-c2-virt/crates/wasi-c2/src/virt/pipe.rs#L375

Reply:
From my perspective, the granularity of various traits in this area is something we're still figuring out. Could you say more about why `Pipe` in your link here uses `Seek`? Elsewhere you mentioned using TCP, QUIC, etc. sockets, however they don't support `Seek`. Things that do implement `Seek` such as files and memory buffers aren't interactive and don't make sense for interactive protocols, which is what one typically does on top of TCP, QUIC, etc. sockets.

Reply:
The use case we wrote ReadPipe and WritePipe for is just to buffering stdin and stdout/err for an instance, rather than communicate between instances. Bi-directional communication is more complex and something I wasn't ready to bite off yet.

Reply:
> Could you say more about why Pipe in your link here uses Seek? Elsewhere you mentioned using TCP, QUIC, etc. sockets, however they don't support Seek.

Yes, that's true; I added it just for my [demo](https://github.com/ueno/wasi-shared-handles/tree/wip/dueno/wasi-c2) of the other branch. On the other hand, I suspect Seek will be desirable in the case with running Wasm programs in TEE, to support transparent encryption of files.